worker/upload.go

```diff
@@ -1069,10 +1069,11 @@ func (s *slabUpload) receive(resp sectorUploadResp) (bool, bool) {
 	// update uploaded sectors
 	s.numUploaded++
 
-	// release all other candidates for this sector
+	// release the candidate if the upload was redundant
 	for _, candidate := range s.candidates {
-		if candidate.req != nil && candidate.req != req && candidate.req.sector.index == sector.index {
+		if candidate.req == req && sector.isUploaded() {
```

Comment:
This looks off to me. This loop is meant to release all candidates that were working on the same sector, because we are processing the response of the first host to which we successfully uploaded the sector. So I'm struggling to see why we would check if the requests match... also I think `sector.isUploaded()` is redundant because we `finish` the sector on line 1062, so this will always be `true`.

```go
// release all other candidates for this sector
for _, candidate := range s.candidates {
    if candidate.req != nil && candidate.req.sector == sector {
        candidate.req = nil
    }
}
```

Obviously if indeed this was wrong it's concerning tests pass, but that goes without saying.

Reply:
Releasing all candidates was the problem. A candidate should only release itself if it was working on a redundant piece and not everyone else. Otherwise we release candidates before they are done uploading. which caused them to be reused early resulting in a failing test.
The tests probably pass because we never force overdrive in other tests. So we never launch requests after the initial ones.

Reply:
Hm ok. `sector.isUploaded()` is still redundant though and I think the comment is a little confusing maybe? It says we release the candidate if it was redundant, but if it was considered redundant we would've exited already on line 1059. 

I'm hesitant to approve the change because it does not feel right. Are you positive your overdrive candidates will release themselves in all scenarios? I think if the request is sitting in the (slow) uploader's queue, it won't get handled properly if we find the sector it's attached to is already done uploading. That's because we simply `continue` there so that uploader will never be selected again as a viable candidate for uploading another sector.

Reply:
Yeah that sounds about right. I moved it up to the `isUploaded` check.